### ISSUE TITLE
WordPress.WhiteSpace.OperatorSpacing.NoSpaceAfter doesn't like JS regex patterns

### ISSUE THREAD

**Issue by GaryJones** [WordPress-Coding-Standards/WordPress-Coding-Standards#1307]
`WordPress.WhiteSpace.OperatorSpacing.NoSpaceAfter` complains about the lack of whitespace in a JavaScript regex pattern.

```js
function ga_skiplinks() {
	'use strict';
	var element = document.getElementById( location.hash.substring( 1 ) );

	if ( element ) {
		// WordPress.WhiteSpace.OperatorSpacing.NoSpaceAfter complains about the next line:
		if ( ! /^(?:a|select|input|button|textarea)$/i.test( element.tagName ) ) {
			element.tabIndex = -1;
		}
		element.focus();
	}
}
```

Result:

```
 21 | ERROR | [x] Expected 1 space before "^"; 0 found (WordPress.WhiteSpace.OperatorSpacing.NoSpaceBefore)
 21 | ERROR | [x] Expected 1 space before "?"; 0 found (WordPress.WhiteSpace.OperatorSpacing.NoSpaceBefore)
 21 | ERROR | [x] Expected 1 space before "|"; 0 found (WordPress.WhiteSpace.OperatorSpacing.NoSpaceBefore)
 21 | ERROR | [x] Expected 1 space before "|"; 0 found (WordPress.WhiteSpace.OperatorSpacing.NoSpaceBefore)
 21 | ERROR | [x] Expected 1 space before "|"; 0 found (WordPress.WhiteSpace.OperatorSpacing.NoSpaceBefore)
 21 | ERROR | [x] Expected 1 space before "|"; 0 found (WordPress.WhiteSpace.OperatorSpacing.NoSpaceBefore)
 21 | ERROR | [x] Expected 1 space before "/"; 0 found (WordPress.WhiteSpace.OperatorSpacing.NoSpaceBefore)
```

**Comment by GaryJones** [comment]
The WPCS [sniff](https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/blob/10ab802992c5f8b06d85d9bebd596cecf8948ffd/WordPress/Sniffs/WhiteSpace/OperatorSpacingSniff.php) is very light, as we're using the parent class.

Reported upstream: https://github.com/squizlabs/PHP_CodeSniffer/issues/1915

**Comment by jrfnl** [comment]
I was just about to suggest that ;-)

Its not actually a problem with the sniff, but with the tokenizer. I've left a comment in the upstream issue.
To get round it for now, I'd suggest using an explicit check ` === false` instead of using the boolean not operator `!` , as that's the root of the problem.

There is nothing we can do on the WPCS side about this, so I suggest we close the issue here.

**Comment by jrfnl** [comment]
Just FYI for anyone coming across this issue: a fix for this has been merged upstream and will be included in the PHPCS 3.3.0 release.
